torch_dicom/preprocessing/datamodule.py

```diff
@@ -0,0 +1,352 @@
+        if self.boxes_filename:
+            if (traces_path := target / self.boxes_filename).exists():
+                dataset = BoundingBoxMetadata(dataset, traces_path, extra_keys=self.boxes_extra_keys)
+            else:
+                raise FileNotFoundError(traces_path)  # pragma: no cover
+
+        return dataset
+
+    def create_sampler(
+        self,
+        dataset: ImagePathDataset,
+        example_paths: List[Path],
+        root: Path,
+        mode: Mode,
+    ) -> Sampler[int]:
+        """
+        Creates a sampler for the given dataset. By default, a random sampler is created for the training dataset
+        and a sequential sampler is created for the validation and test datasets.
+
+        Args:
+            dataset: The dataset for which the sampler is to be created.
+            example_paths: The paths to the examples in the dataset.
+            root: The root path of the dataset.
+            mode: The mode of the dataset. One of "train", "val", or "test".
+
+        Returns:
+            The created sampler for the dataset.
+        """
+        return RandomSampler(dataset) if mode == Mode.TRAIN else SequentialSampler(dataset)
+
+    def create_batch_sampler(
+        self,
+        dataset: ConcatDataset,
+        sampler: Sampler,
+        example_paths: List[Path],
+        roots: List[Path],
+        mode: Mode,
+    ) -> Optional[BatchSampler]:
+        """
+        Creates a batch sampler for aggregate dataset. By default returns ``None``, meaning
+        the default batch sampler.
+
+        Args:
+            dataset: The :class:`ConcatDataset` for which the sampler is to be created.
+            sampler: The :class:`Sampler` for the aggregate dataset.
+            example_paths: The paths to the examples in the aggregate dataset.
+            roots: The root paths of the datasets.
+            mode: The mode of the dataset. One of "train", "val", or "test".
+
+        Returns:
+            The created batch sampler for the dataset.
+        """
+        return  # pragma: no cover
+
+    def _prepare_datasets(
+        self,
+        inputs: Iterable[Path],
+        mode: Mode,
+        **kwargs,
+    ) -> List[MetadataDatasetWrapper]:
+        datasets = [self.create_dataset(inp, mode, **kwargs) for inp in inputs]
+        assert all(isinstance(ds, Sized) for ds in datasets)
+        if not any(isinstance(ds, Sized) and len(ds) for ds in datasets):
+            raise FileNotFoundError("Loaded empty dataset")
+
+        return datasets
+
+    def _prepare_samplers(
+        self,
+        roots: Iterable[Path],
+        datasets: Iterable[MetadataDatasetWrapper],
+        mode: Mode,
+    ) -> List[Sampler]:
+        samplers: List[Sampler] = [
+            self.create_sampler((unwrapped := _unwrap_dataset(dataset)), list(unwrapped.files), root, mode)
+            for dataset, root in zip(datasets, roots)
+        ]
+        return samplers
+
+    def setup(self, stage: Optional[str] = None) -> None:
```

Comment:
Consider `StrEnum` for `stage`?

Reply:
This method is a pytorch-lightning hook so we probably need to conform to the signature they provide.